sdk/ios/Configsum/Configsum/Configsum/Networking/HTTP.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  HTTP.swift
+//  Configsum
+//
+//  Created by Alexandru Savu on 2017-10-04.
+//  Copyright © 2017 Lifesum. All rights reserved.
+//
+
+import Foundation
+
+typealias HTTPCompletion = (Result<Data>) -> Void
+
+protocol HTTP {
+    var logger: Logger { get }
+    var router: Router { get }
+    func put(payload: String, completion: @escaping HTTPCompletion)
+    init(environment: Environment)
+}
+
+fileprivate struct Constants {
+    static let successCodes = 200..<299
+}
+
+extension HTTP {
+    internal func execute(request: URLRequest, payload: String, completion: @escaping HTTPCompletion) {
+        let session = URLSession.shared
+        //TODO Add logger
+        let task = session.dataTask(with: request) { (data: Data?, response:URLResponse?, error:Error?) in
+            completion(Result {
+                if let error = error { throw error }
+                guard let data = data else { throw HTTPError.unknown }
+                guard let httpResponse = response as? HTTPURLResponse else {
+                    throw HTTPError.unknown
+                }
+                guard self.isStatusCodeValid(httpResponse: httpResponse) else {
+                    //TODO Better error handeling
+                    throw HTTPError(status: httpResponse.statusCode, responseBody: "")
+                }
+                return data
+            })
+        }
+        task.resume()
+    }
+}
+
+//Helper functions
+extension HTTP {
+    fileprivate func isStatusCodeValid(httpResponse: HTTPURLResponse) -> Bool {
+        return Constants.successCodes.contains(httpResponse.statusCode)
```

Comment:
We can use pattern matching here, i.e. `return successCodes ~= httpResponse.statusCode`. Doesn't really matter :)